crates/app/src/tags.rs

```diff
@@ -0,0 +1,156 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{assert_repo, RepoStore, TagStore};
+
+use std::sync::Arc;
+
+use drawbridge_jose::jws::Jws;
+use drawbridge_store::{
+    Create, CreateError, CreateFromReaderError, Get, GetError, GetToWriterError, Keys,
+};
+use drawbridge_type::tag::{Entry, Name};
+use drawbridge_type::{repository, tree, RequestMeta};
+
+use axum::body::Body;
+use axum::extract::RequestParts;
+use axum::http::{Request, StatusCode};
+use axum::response::IntoResponse;
+use axum::{Extension, Json};
+use futures::TryStreamExt;
+
+pub async fn query(
+    Extension(repos): Extension<Arc<RepoStore>>,
+    Extension(tags): Extension<Arc<TagStore>>,
+    Extension(repo): Extension<repository::Name>,
```

Comment:
What is this syntax with the function name around the parameter value?

Reply:
https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_structures.html